File: LogicOverSyntaxCodes/Searching/JavaCodes/Leetcode69.java
class Solution {
   public static int mySqrt(int x) {
        int start=0,end=x;

        while(start<=end){
            int mid=start+(end-start)/2;
            long value=(long) mid*mid;
            if(value==x){
                return mid;
            }
            if(value>x){
                end=mid-1;
            }
            else{
                start=mid+1;
            }
        }

        return end;
    }

}

File: Searching/PythonCodes/Leetcode69.py
class Solution(object):
    def mySqrt(self, x):
        start, end = 0, x

        while start <= end:
            mid = start + (end - start) // 2
            value = mid * mid
            if value == x:
                return mid
            if value > x:
                end = mid - 1
            else:
                start = mid + 1

        return end
